packages/hash/frontend/src/components/layout/MainContentWrapper.tsx

```diff
@@ -1,13 +1,75 @@
+import { IconButton, Fade, Box, Tooltip, styled } from "@mui/material";
 import { FunctionComponent } from "react";
+import { SIDEBAR_WIDTH } from "../../theme/components/navigation/MuiDrawerThemeOptions";
 
-import styles from "../../pages/index.module.scss";
+import { SidebarToggleIcon } from "../icons";
+import { HEADER_HEIGHT } from "./PageHeader/PageHeader";
 import { PageSidebar } from "./PageSidebar/PageSidebar";
+import { useSidebarContext } from "./SidebarContext";
+
+const Main = styled("main", {
+  shouldForwardProp: (prop) => prop !== "sidebarOpen",
+})<{
+  sidebarOpen?: boolean;
+}>(({ theme, sidebarOpen }) => ({
+  height: `calc(100vh - ${HEADER_HEIGHT}px)`,
+  overflowY: "scroll",
+  flexGrow: 1,
+  padding: "60px 120px 0 120px",
+  transition: theme.transitions.create("margin", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  marginLeft: `-${SIDEBAR_WIDTH}px`,
+  ...(sidebarOpen && {
+    transition: theme.transitions.create("margin", {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginLeft: 0,
+  }),
+}));
 
 export const MainContentWrapper: FunctionComponent = ({ children }) => {
+  const { openSidebar, sidebarOpen } = useSidebarContext();
+
   return (
-    <div className={styles.MainWrapper}>
+    <Box
+      sx={{
+        display: "flex",
+        position: "relative",
+      }}
+    >
       <PageSidebar />
-      <main className={styles.MainContent}>{children}</main>
-    </div>
+      <Fade in={!sidebarOpen}>
+        <Tooltip title="Open Sidebar">
+          <IconButton
```

Comment:
@MaggieAppleton correct me if I'm wrong, but my guess is we want these icons to be at the same height:

https://user-images.githubusercontent.com/42802102/156800315-399f74af-00b7-4765-8fb8-a1efa06f6091.mov